addon/components/inputs/autocomplete.js

```diff
@@ -50,14 +89,36 @@ export default SelectInput.extend({
   parseValue (data) {
     return data
   },
+
+  // == Actions ===============================================================
   actions: {
-    checkIfEmptyfilter (filterValue) {
-      this.set('filter', filterValue)
+    onInput (filterValue) {
+      const focused = this.get('_focused')
+
+      this.setProperties({
+        filter: filterValue,
+        _isTyping: focused
```

Comment:
Interested to know why it's necessary to keep typing state. Is there no alternative?

Reply:
I probably just need to get a better understanding of the bug you're fixing.

Reply:
Ya I'll add some screenshots/gifs tomorrow to better explain the problem

Reply:
Well the original problem when you have a selectedItem and backspace to an empty filter it would.
![2018-07-19 11_13_08](https://user-images.githubusercontent.com/9057680/42951681-c3ad404e-8b44-11e8-9555-0ee947e3af63.gif)

So if I can keep track of typing I know that I shouldn't set the filter when the user is just backspacing all the way
![2018-07-19 11_01_52](https://user-images.githubusercontent.com/9057680/42952114-b9798b2c-8b45-11e8-8ef1-e4d0b7c850b7.gif)